Updated Use Cases (#44)
AY2425S2-CS2103T-T16-3/tp: docs/DeveloperGuide.md

| `* *`    | new user | import my list of applicant's contact               | seamlessly migrate data from using one device to this another                  |
| `*`      | new user | play around with sample data                        | gain more familiarity with using the application                               |

### Use Cases

**Use case: UC01 - Listing applicants**

**MSS**
1. User requests the list of applicants
2. QuickHire shows the list of applicants

   Use case ends.

**Extensions**

> * 1a. The list is empty.

Reviewer: ```suggestion
* 2a. The list is empty.
```

Reviewer 2: I think for this one. What can we do if the list is empty is to notify the user. For example:
```
2a. The list is empty
    2a1.Notify user about the empty list
```

  Use case ends.

---

**Use case: UC02 - Adding an applicant**

**MSS**
1. User requests to add an applicant
2. QuickHire adds a new applicant

   Use case ends.

**Extensions**

* 2a. Duplicate applicant.
   * 2a1. QuickHire shows an error message.

  Use case ends.

---

**Use case: UC03 - Delete an applicant**

**MSS**

> 1.  User requests to list applicants
> 2.  QuickHire shows a list of applicants
> 3.  User requests to delete a specific applicant in the list
> 4.  QuickHire deletes the person

Reviewer: ```suggestion
1.  User lists applicants (UC01)
2.  User requests to delete a specific applicant in the list
3.  QuickHire deletes the person
```

Reviewer 2: I think this suggestion looks good. To avoid confusion, Harris's suggestions on  the index of the extension in later comments is based on his new update of UC03.

Author: Thanks for the suggestion

> * 3a. The given index is invalid.

Reviewer: ```suggestion
* 2a. The given index is invalid.
```

>     * 3a1. QuickHire shows an error message.

Reviewer: ```suggestion
    * 2a1. QuickHire shows an error message.
```

      Use case resumes at step 2.
      

> * 3b. The given parameters are invalid.
>   
>     * 3b1. QuickHire shows an error message.

Reviewer: ```suggestion
* 2b. The given parameters are invalid.
  
    * 2b1. QuickHire shows an error message.
```
